src/lib/service/init/template.js

```diff
@@ -0,0 +1,176 @@
+	inquirer = require('inquirer'),
+	path = require('path'),
+
+	questions = require('../../util/questions'),
+	validators = require('../../util/validators'),
+
+	AVAILABLE_TEMPLATES = 'availableTemplates',
+	CLI_SELECTED_TEMPLATE = 'argv.template',
+	CONFIGURATION_FILE = '.config.json',
+	QUESTION = 'Select app to create:',
+	SELECTED_TEMPLATE = 'selectedTemplate.folder',
+	SELECTED_TEMPLATE_CONFIGURATION_FILE = 'selectedTemplate.configuration.file',
+	SELECTED_TEMPLATE_CONFIGURATION_FILE_EXTENSIONS = 'selectedTemplate.configuration.file.extends',
+	SELECTED_TEMPLATE_EXTENSION_CONFIGURATION_FILES = 'selectedTemplate.configuration.extensions',
+	SELECTED_TEMPLATE_PATH = 'selectedTemplate.fullPath',
+	TEMPLATE_FOLDER = 'templateFolder';
+
+function isDirectory(source) {
+	return fs.lstatSync(source).isDirectory();
+}
+
+/**
+ * Finds all the templates in the default Orizuru templates folder.
+ * @instance
+ * @param {object} config - The configuration object passed through the process.<br/>This is mutable and should contain the following properties for this part of the process:
+ * @param {string} config.templateFolder - The folder to search for templates.
+ * @returns {object} config - The modified configuration object.
+ */
+function findTemplates(config) {
+
+	const source = _.get(config, TEMPLATE_FOLDER);
+	return fs.readdir(source)
+		.then(files => _.filter(files, file => isDirectory(path.resolve(source, file))))
+		.then(choices => _.set(config, AVAILABLE_TEMPLATES, choices));
+
+}
+
+/**
+ * Reads the template configuration file.
+ */
+function readTemplateConfigurationFile(config, template) {
+
+	const
+		source = _.get(config, TEMPLATE_FOLDER),
+		templatePath = path.resolve(source, template),
+		configFilePath = path.resolve(templatePath, CONFIGURATION_FILE);
+
+	return fs.readJson(configFilePath);
+
+}
+
+/**
+ * Reads the selected template configuration file.
+ */
+function readSelectedTemplateConfigurationFile(config) {
+	const selectedTemplate = _.get(config, SELECTED_TEMPLATE);
+	return readTemplateConfigurationFile(config, selectedTemplate);
+}
+
+/**
+ * Read extension configuration files for the selected template.
+ */
+function readExtensionConfigurationFiles(config) {
+
+	const
+		extensions = _.get(config, SELECTED_TEMPLATE_CONFIGURATION_FILE_EXTENSIONS),
+		extensionTemplateConfigurationFiles = _.map(extensions, extension => {
+			return readTemplateConfigurationFile(config, extension)
+				.then(configuration => ({
+					[extension]: configuration
+				}));
+		});
+
+	return Promise.all(extensionTemplateConfigurationFiles);
+
+}
+
+/**
+ * Reads all the configuration files for the selected template.
+ */
+function readAllConfigurationFiles(config) {
+
+	return readSelectedTemplateConfigurationFile(config)
+		.then(configurationFile => _.set(config, SELECTED_TEMPLATE_CONFIGURATION_FILE, configurationFile))
+		.then(config => {
+			return readExtensionConfigurationFiles(config)
+				.then(contents => {
```

Comment:
It's a bit too soon for Christmas trees 🎅 
I presume that `config` on line 122 is a reference to the same object as line 118. In which case you can de-nest the then on line 124.
```javascript
return readSelectedTemplateConfigurationFile(config)
    .then(configurationFile => {
        _.set(config, SELECTED_TEMPLATE_CONFIGURATION_FILE, configurationFile);
    })
    .then(() => {
        return readExtensionConfigurationFiles(config);
    })
    .then(contents => {
        const mergedContents = _.assign.apply(_, contents);
        return _.set(config, SELECTED_TEMPLATE_EXTENSION_CONFIGURATION_FILES, mergedContents);
    });
```

Reply:
Going to leave this one